calculator/models.py

```diff
@@ -1,3 +1,78 @@
 from django.db import models
+from django.utils import timezone
 
-# Create your models here.
+
+class Loans(models.Model):
+
+    amount = models.FloatField()
+    term = models.PositiveIntegerField()
+    rate = models.FloatField()
+    date = models.DateTimeField(default=timezone.now)
+    created_loans = 0
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.created_loans += 1
+
+    @property
+    def url_id(self):
+        long_id = f'{self.created_loans:015d}'
+        return '{}-{}-{}-{}'.format(long_id[0:3], long_id[3:7], long_id[7:11], long_id[11:16])
+
+    @property
+    def installments(self):
+        r = self.rate/12
+        return (r + r / (((1 + r) ** self.term) - 1))*self.amount
+
+    @property
+    def balance(self):
+        selection = Payments.objects.filter(loan_id=self.id, paid='made')
```

Comment:
here it is possible to use the default django form relationship `self.payment_set.all`